addons/purchase/models/purchase.py

```diff
@@ -413,6 +413,10 @@ def action_view_invoice(self):
             'default_company_id': self.company_id.id,
             'default_purchase_id': self.id,
         }
+        # Invoice_ids may be filtered depending on the user. To ensure we get all
+        # invoices related to the purchase order, we read them in sudo to fill the
+        # cache.
+        self.sudo()._read(['invoice_ids'])
```

Comment:
move this one in a specific commit since it's a bugfix ?

Reply:
do you know if there's actually a record rule needing this patch to work in our standard addons?

Reply:
so it's the  "Personal Invoice Lines" RR right?

Reply:
add a test, edit the new RR and make it so that po users can only see their own documents -> it'll prove this patch is right